students/csimmons/lesson04/mailroom.py

```diff
@@ -37,93 +39,96 @@ def print_donors(donors):
     print('\n')
 
 def exist_donor(response, donors):
-    gift = input(gift_prompt)   
-    f_gift = float(gift)
-    f_response = response.title()
-    for i in range(len(donorlist)):
-        if f_response == donorlist[i][0]:
-            donorlist[i][1].append(f_gift)
-    generate_thankyou(f_response, f_gift)
+    gift = int(input(gift_prompt))
+    response = response.title()
+    for donor in donors:
+        if response == donor:
+            donorlist_dict.setdefault(donor, []).append(gift)
+    generate_thankyou(response, gift)
 
 def new_donor(response):    
-    f_response = response.title()
-    print(('\n{} is a new donor!').format(f_response))
-    gift = input(gift_prompt)
-    f_gift = float(gift)
-    new_donor = tuple([f_response,[ ]])
-    donorlist.append(new_donor)
-    get_length = len(donorlist)-1
-    donorlist[get_length][1].append(f_gift)
-    generate_thankyou(f_response, f_gift)
+    response = response.title()
+    print(('\n{} is a new donor!').format(response))
+    gift = int(input(gift_prompt))
+    donorlist_dict[response] = [gift]
+    generate_thankyou(response, gift)
 
 def print_donorlist(all_info):
     header1 = '{:20}{:1}{:15}{:1}{:10}{:1}{:12}'.format('\n''Donor Name ', '|', ' Total Given ', '|', ' Num Gifts ', '|', ' Average Gift ')
     header2 = ('_ ' * 32) +'\n'
     info_row = '{dname:<20s}$ {total:>13,.2f} {gifts:^10d}  $ {avg:>12,.2f}'.format
     print(header1)
     print(header2)
-    for i in range(len(all_info)):
-        print(info_row(dname=all_info[i][0], total=all_info[i][1], gifts=all_info[i][2], avg=all_info[i][3]))
+    for idx, info in enumerate(all_info):
+        name1 = all_info[idx][0]
+        total1 = all_info[idx][1]
+        gifts1 = all_info[idx][2]
+        avg1 = all_info[idx][3]
+        print(info_row(dname=name1, total=total1, gifts=gifts1, avg=avg1))
     print('\n')
 
-def generate_thankyou(f_response, f_gift):
-    print("""\n
-    Dear {},
-    We would like to thank you for your extremely generous donation 
-    of ${:,.2f} to the Anonymous Charity of Seattle.\n
-    Sincerely,
-    CA Simmons \n""".format(f_response, f_gift))
+def generate_thankyou(response, gift):
+    print(('\nDear {},\n\n'
+        'We would like to thank you for your extremely generous donation\n'
+        'of ${:,.2f} to the Anonymous Charity of Seattle.\n\n'
+        'Sincerely,\n\n'
+        'CA Simmons \n'.format(response, gift)))
 
-def send_thankyou(donorlist):
-    donors = list(map(lambda x:x[0], donorlist))
+def send_thankyou(donorlist_dict):
+    donors = list(donorlist_dict.keys())
```

Comment:
why convert your dict to list? we want to work with dictionaries in this lesson - they are so much easier to work with 🙂